### PR TITLE
JsBindings: Add tests for bindings with custom implementations

### PR DESCRIPTION
**Pull request by frericp**
These tests cover bindings with custom implementations, and also validate some more generic approaches that we take with the bindings (e.g. the way we bind optional parameters, or deal with exceptions).

Update #1184

### CODE REVIEW

File: source/JsMaterialX/test/customBindings.spec.js [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        resolver.setGeomNameSubstitution(gnTestKeys[1], gnTestData[gnTestKeys[1]]);
+        const gnSubs = resolver.getGeomNameSubstitutions();
+        expect(gnSubs).to.be.instanceof(Object);
+        expect(Object.keys(gnSubs).length).to.equal(2);
+        expect(gnSubs).to.deep.equal(gnTestData);
+    });
+
+    it('getShaderNodes', () => {
+        const doc = mx.createDocument();
+        const mtlxStrs = getMtlxStrings(['MaterialBasic.mtlx'], '../../../resources/Materials/Examples/Syntax');
```
Comment (kohakukun) [inline source/JsMaterialX/test/customBindings.spec.js:168]: Just some nit but, could we add a const variable with the resources path

File: source/JsMaterialX/test/customBindings.spec.js [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+        // Document.getVersionIntegers
+        versionStringArr.pop();
+        const doc = mx.createDocument();
+        const docVersion = doc.getVersionIntegers();
+        expect(docVersion).to.be.instanceof(Array);
+        expect(docVersion.length).to.equal(2);
+        expect(docVersion).to.deep.equal(versionStringArr);
+
+        // InterfaceElement.getVersionIntegers (via NodeDef)
```
Comment (kohakukun) [inline source/JsMaterialX/test/customBindings.spec.js:217]: Leftovers?

File: source/JsMaterialX/test/customBindings.spec.js [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        shaderNodes = mx.getShaderNodes(matNode, mx.DISPLACEMENT_SHADER_TYPE_STRING);
+        expect(shaderNodes).to.be.instanceof(Array);
+        expect(shaderNodes.length).to.equal(1);
+        expect(shaderNodes[0].getType()).to.equal(mx.DISPLACEMENT_SHADER_TYPE_STRING);
+        shaderNodes = mx.getShaderNodes(matNode, 'bogus');
+        expect(shaderNodes).to.be.instanceof(Array);
+        expect(shaderNodes.length).to.equal(0);
+
+        // TODO: Couldn't find a case / sample file where an input node has a target definition
+        // Should filter shader nodes based on the given optional target type
```
Comment (kohakukun) [inline source/JsMaterialX/test/customBindings.spec.js:190]: Cannot you create them in place? like `doc.addTargetDef()`

Comment (frericp) [reply source/JsMaterialX/test/customBindings.spec.js:190]: Hm, not sure if this is what the flag is related to, but I'll give it a try! 👍

**Review by kohakukun — APPROVED** [review]

**Pull request merged**